### PR TITLE
Update to Linebender CI configuration

### PR DESCRIPTION
**Pull request by waywardmonkeys**

### CODE REVIEW

**Comment by waywardmonkeys** [comment]
This is a draft to see what fails other than typos, copyright, clippy.

**Comment by waywardmonkeys** [comment]
@xStrom @DJMcNab Let's do this!

File: .github/workflows/ci.yml
Comment (DJMcNab) [inline .github/workflows/ci.yml:1]: Is it worth trying to track `no_std` for this crate?

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,4 @@
 target
 **/*.rs.bk
```
Comment (DJMcNab) [inline .gitignore:2]: I don't think this meaningful to still have https://github.com/rust-lang/rustfmt/issues/1114
```suggestion
```

**Review by DJMcNab — APPROVED** [review]

**Comment by DJMcNab** [comment]
When you're ready to land this, let me know and I'll do the repository setup beforehand.

File: .github/workflows/ci.yml
Comment (waywardmonkeys) [reply .github/workflows/ci.yml:1]: Not prior to this landing.  I've had it partially working in the past but tossed those changes as it was too much to do before other stuff was done. And I think there was some sort of complication, but I forget now.

File: .gitignore (hunk 1 above)
Comment (waywardmonkeys) [reply .gitignore]: Done.

**Comment by waywardmonkeys** [comment]
> When you're ready to land this, let me know and I'll do the repository setup beforehand.

Ping me on Zulip anytime after 45 minutes from now.

**Pull request merged**